### PR TITLE
Java: Fix TimeSkill unit test

### PR DESCRIPTION
**Pull request by brunoborges**

### CODE REVIEW

**Comment by johnoliver** [comment]
I would modify the method `assertEqualsIgnoreCase` to be something like `assertEqualIgnoreRegionalSuffix` and then remove the '.'s . We already ignore case to account for regional changes, other option is to bring back #1821 that should fix it once and for all

**Comment by brunoborges** [comment]
This PR now ensures Locale is passed on to all methods in TimeSkill. It also uses system's default as locale, instead of `en_US` as previously coded.

The class `LocaleParser` may be used by other skills, core, or 3rd-party provided, to ensure compatibility.

File: java/semantickernel-core-skills/src/main/java/com/microsoft/semantickernel/coreskills/TimeSkill.java [hunk 1]
```diff
@@ -219,9 +289,15 @@ public String hourNumber() {
     @DefineSKFunction(
             name = "daysAgo",
             description = "Get the date of offset from today by a provided number of days")
-    public static String daysAgo(String days) {
+    public static String daysAgo(
+            String days,
```
Comment (johnoliver) [inline java/semantickernel-core-skills/src/main/java/com/microsoft/semantickernel/coreskills/TimeSkill.java:293]: probably needs annotations on days, make non static

File: java/semantickernel-core-skills/src/main/java/com/microsoft/semantickernel/util/LocaleParser.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+     * tag. If the input is "en_US", it will return a Locale based on the language and country code.
+     * Otherwise, it will return a Locale based on the input.
+     *
+     * <p>'en' 'US' used as example above.
+     *
+     * @param locale the locale in String text.
+     * @return
+     */
+    public static final Locale parseLocale(String locale) {
+        if (localeCache.containsKey(locale)) {
```
[the remark this answers is not in the document]
Comment (brunoborges) [reply java/semantickernel-core-skills/src/main/java/com/microsoft/semantickernel/util/LocaleParser.java:34]: I don't think an application would end up using all possible locales. But even if they do, the amount of possible locales is limited, and not huge.

And as much as it sounds premature, I don't see a problem in caching Locale objects.

[the remark this answers is not in the document]
Comment (dsgrieve) [reply java/semantickernel-core-skills/src/main/java/com/microsoft/semantickernel/util/LocaleParser.java:33]: Why not Locale#lookup?

**Review by dsgrieve — APPROVED** [review]

**Comment by brunoborges** [comment]
@dsgrieve how can we effectively use `Locale.lookup` as a replacement for a HashMap caching?

**Review by johnoliver — DISMISSED** [review]

**Review by markwallace-microsoft — DISMISSED** [review]

**Comment by dsgrieve** [comment]
> @dsgrieve how can we effectively use `Locale.lookup` as a replacement for a HashMap caching?

My thought was that one of the Locale #lookup methods would do the magic of matching e.g. "en-gb" to en-GB and you wouldn't have to do all of that if-block logic. Indeed, it may not be necessary since Locale.forLanguageTag("en-gb") does yield en-GB (at least on java 17). Also, I think your code will fail if there is a variant e.g. en-us-posix. It might be that all you need to do is Locale.lookup(locale.replace("_","-")). 

I agree with John that the caching seems unnecessary. 

Final point, just to be picky, you could use Map#computeIfAbsent instead of rockin' it old school.

**Review by dsgrieve — APPROVED** [review]

**Comment by brunoborges** [comment]
@dsgrieve if the locale passed on is `en-us-posix`, it should work fine with the new database in JDK 9+.

Granted, the Locale thing is unlikely to be 100% trouble-free because we are dealing with meaningful changes between JDK 8 and JDK 9+, and the end user may even change JDK 9+ runtime to behave like JDK 8.

Worse case scenario, the end user of the library may have to roll their own TimeSkill to deal with Locale/Timezones however they want, either by inheriting this class, or by writing their own TimeSkill.

**Review by markwallace-microsoft — COMMENTED** [review]
LGTM

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**